src/osd/PGLog.cc

```diff
@@ -942,8 +942,7 @@ void PGLog::_write_log_and_missing(
       if (!missing.is_missing(obj, &item)) {
 	to_remove.insert(key);
       } else {
-	uint64_t features = missing.may_include_deletes ? CEPH_FEATURE_OSD_RECOVERY_DELETES : 0;
-	encode(make_pair(obj, item), (*km)[key], features);
+	encode(make_pair(obj, item), (*km)[key], CEPH_FEATURES_SUPPORTED_DEFAULT);
```

Comment:
I think we should consider the side-effect of using  CEPH_FEATURES_SUPPORTED_DEFAULT flag here, maybe something like CEPH_FEATURE_OSD_RECOVERY_DELETES | SERVER_OCTOPUS is much reasonable here. The modification here break the original logic of missing.may_include_deletes; Otherwise LGTM

Reply:
can we just use a check for SERVER_OCTOPUS?

Reply:
emm, the background of introducing the  flag CEPH_FEATURE_OSD_RECOVERY_DELETES in the commit a51d5fd9a11e852361df5e3e90d04710a8f19cb4,  @jdurgin; I think it is a versioned flag to indicate whether the pg_log supporting deletes operation. Maybe we can remove this flag in feature release since all recent release supporting this features now. But I'm afraid that currently we should still preserve the flag here until sometime we raise a proposal to clear legacy feature flags, right?

Reply:
![pg_missing_item-encode](https://user-images.githubusercontent.com/19184136/133041776-7bb34add-45c7-4bff-ad0c-99d265c4620a.png)

this codes picture show that the pg_missing_item::enocde() function only check the SERVER_OCTOPUS flag.

Since features are only used for version control in this call, and pg_missing_item only check SERVER_OCTOPUS flag, I also think it is more appropriate to only use SERVER_OCTOPUS.

@neha-ojha

Reply:
yeah, seems LGTM after octopus, and legacy flags can be cleared.